### ISSUE TITLE
Homebrew completions nullglob error

### ISSUE THREAD

**Issue by oschrenk** [fish-shell/fish-shell#3344]
**fish version installed** *(`fish --version`)*:

fish, version 2.3.0-224-g57f289

**OS/terminal used**: 

macOS Sierra, 10.12, 16A313a

## Reproduction steps

1. Have lastest homebrew, via `brew update`. 
2. Start typing `brew info n`.
3. Press tab to invoke autocompletion
4. See autocompletion error

## Expected results

Present autocompletions

## Actual results

```
No matches for wildcard '$formuladir/*.rb'.  (Tip: empty matches are allowed in 'set', 'count', 'for'.)
/usr/local/Cellar/fish/HEAD/share/fish/completions/brew.fish (line 3):     ls $formuladir/*.rb | sed 's/.rb$//' | sed "s|^$formuladir||"
                                                                              ^
in function '__fish_brew_formulae'
        called on standard input

in command substitution
        called on standard input
```

## Workaround, suggested fix

I  fixed it by changing a function in `/usr/local/Cellar/fish/HEAD/share/fish/completions/brew.fish`

from

```
function __fish_brew_formulae
    set -l formuladir (brew --repository)/Library/Formula/
    # __fish_complete_suffix .rb
    ls $formuladir/*.rb | sed 's/.rb$//' | sed "s|^$formuladir||"
end
```

to

```
function __fish_brew_formulae
    set -l formuladir (brew --repository)/Library/Taps/
    find (brew --repository)/Library/Taps -iname '*.rb' | rev | cut -d "/" -f1 | rev | sed 's/.rb$//'
end
```

**Comment by faho** [comment]
Really, this should be switched over to `string`. The `ls` is completely unnecessary.

Can you try:

```fish
function __fish_brew_formulae
    set -l formuladir (brew --repository)/Library/Formula/
    # __fish_complete_suffix .rb
    string replace -r '.rb$' '' -- "$formuladir"/*.rb | string replace "$formuladir" ''
end
```

>    set -l formuladir (brew --repository)/Library/Taps/

Why did you change the "Formula" to "Taps"? I'm not really familiar with homebrew if you haven't guessed by now.

**Comment by oschrenk** [comment]
Mainly I switched to `Taps` directory because the old one doesn't exist for me anymore. 

```
ls (brew --repository)/Library/Formula/
ls: /usr/local/Library/Formula/: No such file or directory
```
I guess Homebrew is transitioning to a more modular model, delivering formulas, even the core ones, within a tap (homebrew name for a collection of formulas). I found the formulas in the `(brew --repository)/Library/Taps/` dir but there is not so much consistence in how they are organized within a tap and that is why I'm using a find command.

I'm not that familiar with the `string` command but I can look what I can do to improve the `find` command.

A concern I have is that just switching to Taps dir breaks autocompletion for users that didn't update their homebrew installation in a while.

**Comment by oschrenk** [comment]
I think it's actuall fixed with a superior solution to mine:

https://github.com/fish-shell/fish-shell/commit/5dd959070ce9b1af98634b8731e9ef7b0563c637

**Comment by oschrenk** [comment]
Yeah. Fixed in 

fish, version 2.3.1-494-geeb42f5

Sorry for the noise. Closing.

**Comment by faho** [comment]
Yes, fixed via #3309.
